src/control/vendor/github.com/daos-stack/go-spdk/spdk/src/nvme_control.c

```diff
@@ -193,22 +175,29 @@ collect(struct ret_t *ret)
 			return;
 		}
 
-		cdata = spdk_nvme_ctrlr_get_data(ns_entry->ctrlr);
+		ns_ctrlr = spdk_nvme_ns_get_ctrlr(ns_entry->ns);
```

Comment:
Why not use `ns_entry->ctrlr` here instead of creating the `ns_ctrlr` variable. From `register_ns()` shouldn't you already have access to the controller associated to `ns_entry->ns`?

Reply:
can do, had a moment when I thought it would be better to use the built-in methods but it doesn't really make sense so I will revert